src/components/datePicker/DatePickerReducers.ts

```diff
@@ -36,27 +38,37 @@ export const datePickerInitialState: IDatePickerState = {
 export const datePickersInitialState: IDatePickerState[] = [];
 
 const addDatePicker = (state: IDatePickerState, action: IReduxAction<IAddDatePickerPayload>): IDatePickerState => {
+  const mayBeNull = (d: Date) => action.payload.initiallyUnselected ? null : d;
```

Comment:
you should move this function in `FalsyValuesUtils.ts` and add quick UTs for it

Reply:
I don't think this could be useful elsewhere because the criteria used to nullify the value is `action.payload.initiallyUnselected` which is the only place where the payload contains such information.

If I generalize the function by passing the criteria as a parameter, then it just falls down to using a basic ternary operation. I made this function only to avoid repeating the same ternary over and over again.